tests/backend/test_ws_events.py

```diff
@@ -0,0 +1,113 @@
+
+    async def _mocked_api_events_listen(client_ctx, msg):
+        nonlocal backend_client_ctx
+        backend_client_ctx = client_ctx
+        backend_received_cmd.set()
+        return await vanilla_api_events_listen(client_ctx, msg)
+
+    running_backend.backend.apis[HandshakeType.AUTHENTICATED][
+        "events_listen"
+    ] = _mocked_api_events_listen
+
+    events_listen_rep = None
+
+    async with backend_sock_factory(running_backend.backend, alice, keepalive=2) as transport:
+        mock_clock.rate = 1
+        async with trio.open_service_nursery() as nursery:
+
+            async def _cmd():
+                nonlocal events_listen_rep
+                req = {"cmd": "events_listen", "wait": True}
+                raw_req = events_listen_serializer.req_dumps(req)
+                await transport.send(raw_req)
+                events_listen_rep = await transport.recv()
+
+            nursery.start_soon(_cmd)
+
+            # Wait for the connection to be established with the backend
+            with trio.fail_after(1):
+                await backend_received_cmd.wait()
+
+            # The backend should upgrade the websocket connection
+            backend_transport, event = await next_ws_proto_related_event()
+            assert isinstance(event, Request)
+
+            # The client accept the new websocket connection
+            client_transport, event = await next_ws_proto_related_event()
+            assert isinstance(event, AcceptConnection)
+
+            # Now advance time until ping websocket is requested
+            await trio.testing.wait_all_tasks_blocked()
+            mock_clock.jump(2)
+            with trio.fail_after(2):
+                backend_transport1, event = await next_ws_proto_related_event()
+                assert isinstance(event, Ping)
+                client_transport1, event = await next_ws_proto_related_event()
+                assert isinstance(event, Pong)
+                assert client_transport1 is client_transport
+                assert client_transport1 is not backend_transport
+                assert backend_transport is backend_transport1
+
+            # Wait for another ping websocket, just to be sure...
+            await trio.testing.wait_all_tasks_blocked()
+            mock_clock.jump(2)
+            with trio.fail_after(2):
+                backend_transport2, event = await next_ws_proto_related_event()
+                assert isinstance(event, Ping)
+                assert backend_transport1 is backend_transport2
+                client_transport2, event = await next_ws_proto_related_event()
+                assert isinstance(event, Pong)
+                assert client_transport1 is client_transport2
+
+            # send another request in the steam to be sure that will be managed by the backend
+            raw_req = ping_serializer.req_dumps({"cmd": "ping", "ping": "foo"})
+            await transport.send(raw_req)
+            await trio.testing.wait_all_tasks_blocked()
```

Comment:
Instead of this, we could move the nursery so that we wait for it coroutine completion (i.e. `_cmd` given it is the only one scheduled) before moving on:
```python
assert testing_stuff
with trio.open_nursery() as nursery:
    nursery.start_soon(_cmd)
    await do_stuff
    assert testing_stuff
    await transport.send(raw_req)  # Now _cmd is going to receive a response
assert events_listen_rep == {...}
```